Asterix/optics/optical_systems.py

```diff
@@ -760,7 +770,11 @@ def EF_from_phase_and_ampl(self, phase_abb=0., ampl_abb=0., wavelengths=-1.):
 
         entrance_EF = []
         for wavelength in wavelength_vec:
-            entrance_EF.append((1 + ampl_abb) * np.exp(1j * phase_abb * self.wavelength_0 / wavelength))
+            entrance_EF.append(
+                np.asarray((1 + ampl_abb) *
+                           np.exp(1j * phase_abb * self.wavelength_0 / wavelength, dtype=self.dtype_complex)).astype(
+                               self.dtype_complex))
```

Comment:
Again, why the bit definition twice?

Reply:
in case ampl_abb was previously defined as a certain type but we want to change that. 

this line is doing (1+amp)*exp(i*phase) so we need both term of the multiplication to be in the right dimension